.github/workflows/charm4py.yml

```diff
@@ -28,3 +28,44 @@ jobs:
         export PYTHONPATH="$PWD"
         export CHARM4PY_TEST_NUM_PROCESSES=2
         python3 auto_test.py
+
+  macos-arm:
+    runs-on: macos-14
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: build-charm4py
+      run: |
+        git fetch --unshallow # Need full repo for 'git describe' used by setup.py
+        pip3 install setuptools cython cffi greenlet numpy torch torchvision
+        git clone https://github.com/UIUC-PPL/charm charm_src/charm
+        export CHARM_EXTRA_BUILD_OPTS="--enable-error-checking"
+        export CHARM_BUILD_PROCESSES=2
+        export CHARM4PY_BUILD_CFFI=1
+        python3 setup.py build_ext --inplace
+    - name: test-charm4py
+      run: |
+        export PYTHONPATH="$PWD"
+        export CHARM4PY_TEST_NUM_PROCESSES=2
+        python3 auto_test.py
+
+  macos-arm:
```

Comment:
is this x86_64 by any chance?

Reply:
Also, if the code for the CI steps is identical, this could probably be a build matrix.